gcloud-java-storage/src/test/java/com/google/gcloud/storage/ITStorageTest.java

```diff
@@ -0,0 +1,473 @@
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.storage;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import com.google.common.collect.ImmutableList;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Iterator;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ITStorageTest {
+
+  private static Storage storage;
+  private static RemoteGcsHelper gcsHelper;
+
+  private static final String bucket = RemoteGcsHelper.generateBucketName();
+  private static final String CONTENT_TYPE = "text/plain";
+  private static final byte[] BLOB_BYTE_CONTENT = {0xD, 0xE, 0xA, 0xD};
+  private static final String BLOB_STRING_CONTENT = "Hello Google Cloud Storage!";
+
+  @BeforeClass
+  public static void beforeClass() {
+    try {
+      gcsHelper = RemoteGcsHelper.create();
+      storage = StorageFactory.instance().get(gcsHelper.options());
+      storage.create(BucketInfo.of(bucket));
+    } catch (RemoteGcsHelper.GcsHelperException e) {
+      // ignore
+    }
+  }
+
+  @AfterClass
+  public static void afterClass()
+      throws ExecutionException, TimeoutException, InterruptedException {
+    if (storage != null) {
+      if (!RemoteGcsHelper.deleteBucketRecursively(storage, bucket, 5, TimeUnit.SECONDS)) {
+        throw new RuntimeException("Bucket deletion timed out. Could not delete non-empty bucket");
+      }
+    }
+  }
+
+  @Before
+  public void beforeMethod() {
+    org.junit.Assume.assumeNotNull(storage);
+  }
+
+  @Test(timeout = 5000)
+  public void testListBuckets() throws InterruptedException {
+    Iterator<BucketInfo> bucketIterator =
+        storage.list(Storage.BucketListOption.prefix(bucket)).iterator();
+    while (!bucketIterator.hasNext()) {
+      Thread.sleep(500);
+      bucketIterator = storage.list(Storage.BucketListOption.prefix(bucket)).iterator();
+    }
+    while (bucketIterator.hasNext()) {
```

Comment:
Nit, If we don't expect this to be the first one (and think it may have more files with similar prefix) than the loop above may never get it...